server/src/main/java/org/elasticsearch/index/query/RankDocsQueryBuilder.java

```diff
@@ -55,6 +53,17 @@ public RankDocsQueryBuilder(StreamInput in) throws IOException {
         }
     }
 
+    @Override
+    protected void extractInnerHitBuilders(Map<String, InnerHitContextBuilder> innerHits) {
+        if (queryBuilders != null) {
+            for (QueryBuilder query : queryBuilders) {
+                if (query instanceof AbstractQueryBuilder) {
+                    ((AbstractQueryBuilder<?>) query).extractInnerHitBuilders(innerHits);
```

Comment:
Should this be:
```suggestion
                if (query instanceof AbstractQueryBuilder clause) {
                    InnerHitContextBuilder.extractInnerHits(clause, innerHits);
```
?

Reply:
The `InnerHitContextBuilder#extractInnerHits` calls the `extractInnerHitBuilder` method on the `AbstractQueryBuilder` and throws if a different instance type is passed, so, IIUC, seems equivalent . 

Can refactor however to directly call the ` InnerHitContextBuilder.extractInnerHits` as suggested, which also removes the need to check instance type here and throw if needed, as we'd normally do.

Reply:
Updated.